aten/src/ATen/Context.h

```diff
@@ -73,6 +73,9 @@ class TORCH_API Context {
   static bool hasXLA() {
     return c10::impl::hasDeviceGuardImpl(at::DeviceType::XLA);
   }
+  static bool hasLazy() {
+    return c10::impl::hasDeviceGuardImpl(at::DeviceType::Lazy);
```

Comment:
I remember seeing code in the LazyTensor branch to register an XLA device guard: https://github.com/pytorch/pytorch/blob/lazy_tensor_staging/lazy_tensor_core/lazy_tensor_core/csrc/tensor_impl.cpp#L49

Thinking about how this should work now - I think we'll need separate device guard registrations for Lazy and XLA? (and potential future plugins that gets its own dispatch key).

Maybe the different guards can all re-use the same `LtcGuardImpl`, although right now `LtcGuardImpl` hardcodes its device type as XLA: https://github.com/pytorch/pytorch/blob/5cf77682ad1017b6064c1cf3c689b40df814b8b3/lazy_tensor_core/lazy_tensor_core/csrc/tensor_impl.cpp#L18

Reply:
You're right, I plan to adjust the lazy_tensor_staging branch to use `Lazy` instead of `XLA`, I have the patch ready for that. Yes, we'll have separate guard registrations for Lazy and XLA keys.